graph/src/data/query/result.rs

```diff
@@ -4,10 +4,19 @@ use serde::ser::*;
 use super::error::{QueryError, QueryExecutionError};
 use data::graphql::SerializableValue;
 
+fn serialize_data<S>(data: &Option<q::Value>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    SerializableValue(data.as_ref().unwrap_or(&q::Value::Null)).serialize(serializer)
+}
+
 /// The result of running a query, if successful.
-#[derive(Debug)]
+#[derive(Debug, Serialize)]
 pub struct QueryResult {
+    #[serde(skip_serializing_if = "Option::is_none", serialize_with = "serialize_data")]
```

Comment:
Ah cool, serde has an attribute for everything it seems.

Reply:
Yup 😄